core/CMakeLists.txt

```diff
@@ -796,19 +795,21 @@ if (NOT APPLE)
       # to avoid the error "-r and -pie may not be used together". See i#2083.
       set(disable_pie_flag "-no-pie")
     endif ()
-    add_custom_command(TARGET dynamorio_static POST_BUILD
-      COMMAND
-      ${CMAKE_C_COMPILER} ${partial_link_flags} -fPIC
-        -nostartfiles -nodefaultlibs -Wl,-r ${disable_pie_flag}
-        -Wl,--whole-archive "${dynamorio_dot_a}" -Wl,--no-whole-archive
-        -o "${dynamorio_dot_o}" &&
-      ${CMAKE_OBJCOPY} --localize-hidden "${dynamorio_dot_o}" &&
-      ${globalize_pic_thunks}
-      rm -f "${dynamorio_dot_a}" &&
-      ${CMAKE_AR} cr "${dynamorio_dot_a}" "${dynamorio_dot_o}" &&
-      ${CMAKE_RANLIB} "${dynamorio_dot_a}"
+
+    add_custom_command(TARGET dynamorio_static
+      POST_BUILD
+      COMMAND ${CMAKE_COMMAND}
+      ARGS -D lib_fileloc=${locvar_name}
```

Comment:
Below this sets WORKING_DIRECTORY to the lib output dir, yet the locvar_name.cmake is generated wrt the current build dir: it seems like it's going to fail if we use relative dirs.  It seems very fragile.  Wouldn't absolute dirs be more robust and have fewer problems?

Reply:
Do you mean absolute dirs with file(GENERATE) by setting CMP0070 to OLD? I did not want to introduce another 'OLD' behavior policy as the intention of this patch was towards getting rid of them.

Reply:
Looking at CMP0070 (https://cmake.org/cmake/help/v3.12/policy/CMP0070.html) it only applies when we use a relative path.  So I don't see how CMP0070 comes into play at all if we use absolute paths?  I'm suggesting using ${CMAKE_CURRENT_BINARY_DIR}/myfile or something like that.  Then we remove the fragileness related to where the script's working directory is and what today is an implicit hidden parameter, the current directory.  We also remove any need to care about CMP0070 or adjust any path depending on the cmake version.

Reply:
Ok, great. When I did this I didn't realize that file(GENERATE) can take absolute paths. I changed it to abs path and removed CMP0070.